### DISCUSSION TITLE
Is it possible to change the log level while the application runs?

### DISCUSSION THREAD

**Discussion by nemesyssoft** [CocoaLumberjack/CocoaLumberjack#1399] [Q&A] [answered]
In my `AppDelegate.h`, I did define this:

```
#define LOG_LEVEL_DEF ddLogLevel
@import CocoaLumberjack;
static DDLogLevel ddLogLevel = DDLogLevelAll;
```

In `AppDelegate.m`, in `applicationDidFinishLaunching:`, I check if I did save the log level from a previous session and then set it:
```
    NSNumber *logLevel = [defaults objectForKey:SAVED_LOG_LEVEL];
    if (logLevel != nil) {
        switch (logLevel.integerValue) {
            case 0:
                ddLogLevel = DDLogLevelVerbose;
                break;
                
            case 1:
                ddLogLevel = DDLogLevelDebug;
                break;
                
            case 2:
                ddLogLevel = DDLogLevelWarning;
                break;
                
            default:
                ddLogLevel = DDLogLevelError;
                break;
        }
    } else {
        ddLogLevel = DDLogLevelError;
    }

```
But, even though I declared my `ddLogLevel` static variable as `extern ddLogLevel` in all my other classes, my settings view controller doesn't seem to read the value of that static variable as it reads `ddLogLevel` being set to `DDLogLevelAll`.

So, what am I missing?

**Comment by ffried** [comment, marked as answer]
This would mean that you need to import `AppDelegate.h` in every place you use `DDLogXYZ`. And this in turn means that each of those headers define a new `ddLogLevel`. What you need to do is to declare your `ddLogLevel` as `extern` in `AppDelegate.h` and initialize it in `AppDelegate.m`.
Or even better: Create new `Logging.h` and `Logging.m` files that take care of this, removing the need to link `AppDelegate.h` to every place in the app that needs logging 😉

**Reply by nemesyssoft** [reply]
Thanks.
